probe/awsecs/client.go

```diff
@@ -106,45 +219,173 @@ func (c ecsClient) getTasks(taskArns []string) (map[string]*ecs.Task, error) {
 		Tasks:   taskPtrs,
 	})
 	if err != nil {
-		return nil, err
+		log.Warnf("Failed to describe ECS tasks, ECS service report may be incomplete: %v", err)
+		return
 	}
 
 	for _, failure := range resp.Failures {
 		log.Warnf("Failed to describe ECS task %s, ECS service report may be incomplete: %s", failure.Arn, failure.Reason)
 	}
 
-	results := make(map[string]*ecs.Task, len(resp.Tasks))
 	for _, task := range resp.Tasks {
-		results[*task.TaskArn] = task
+		c.taskCache[*task.TaskArn] = newECSTask(task)
 	}
-	return results, nil
 }
 
-// returns a map from task ARNs to service names
-func (c ecsClient) getInfo(taskArns []string) (ecsInfo, error) {
-	servicesChan := make(chan map[string]*ecs.Service)
-	go func() {
-		servicesChan <- c.getServices()
-	}()
+// Evict entries from the caches which have not been used within the eviction interval.
+func (c ecsClient) evictOldCacheItems() {
```

Comment:
I don't think a standard cache would be suitable. Our goal here isn't to keep a certain size budget of things cached for performance, it's to absolutely minimize the number of calls we're making by explicitly managing the cache. Things shouldn't be evicted if they're still in use - in the case of services in particular this will trigger an expensive list-and-describe-all.
The timed eviction logic is only here so we can withstand a given task flickering out of existence due to a restarting container or other temporary weirdness. Otherwise we'd be explicitly only carrying forward the items we used last round.

With this in mind, maybe the name "cache" is misleading - perhaps "known" or even just simply "tasks" and "services"?